src/reactUnusedPropsAndStateRule.ts

```diff
@@ -104,6 +136,109 @@ function walk(ctx: Lint.WalkContext<Options>) {
         return false;
     }
 
+    function inspectPropUsageInObjectBinding(name: ts.ObjectBindingPattern): void {
+        const bindingElements = getObjectBindingData(name);
+        const foundPropNames = Object.keys(bindingElements);
+
+        for (const propName of foundPropNames) {
+            propNames = Utils.remove(propNames, propName);
+        }
+    }
+
+    function lookForReactSpecificArrowFunction(node: ts.TypeReferenceNode): void {
+        const nodeTypeText = node.typeName.getText();
+
+        const isFunctionComponent =
+            nodeTypeText === 'React.SFC' ||
+            nodeTypeText === 'React.FC' ||
+            nodeTypeText === 'React.StatelessComponent' ||
+            nodeTypeText === 'React.FunctionComponent';
```

Comment:
Could you add some a test case for each of these? They probably don't have to be as extensive as the existing ones, but there should be just enough to verify that these strings are in fact checked.

Reply:
I ended up getting overwhelmed at how the tests were organized while tackling this and ended up breaking them down into more specific areas (this also helped show which areas were better covered than others:

<img width="282" src="https://user-images.githubusercontent.com/359239/52979456-0ef77280-33ac-11e9-96d5-e148b992fbb3.png">

The new tests were named (for both the old `SFC`/`StatelessComponent` and the new `FC`/`FunctionComponent` APIs: 

 - `reports-for-explicit-import-shorthand.tsx.lint`
 - `reports-for-explicit-import.tsx.lint`